### ISSUE TITLE
[Screen Reader-Microbit Menu Controls] Micro bit menu controls like Basic, Input are not accessible.

### ISSUE THREAD

**Issue by v-swasha** [Microsoft/pxt#2584]
**User Experience:**
Users with Visual disabilities might not able to access the controls if some elements are not accessible by Keyboard. Like in the scenario micro bit menu controls are not accessible by keyboard.

**Test Environment**
OS: RS2 Version 1703(OS Build 15063.483)
Platform: Edge.
Screen reader: Narrator

**Repro Steps**
1-Navigate to url:- https://makecode.com/
2-Navigate to the Micro bit section element and select Code control given on it.
3-Navigate to the menu controls listed near the Simulator control.
4-Navigate to the controls and verify the issue.

**Actual Result**
Microbit menu controls like Basic, Input are not accessible by Keyboard.

**Expected Result**
Microbit menu controls should be accessible by keyboard.

**Note**
1) Microbit Menu controls are accessible after editing the search editable field.
2) Controls under Advance Expand button are not even accessible after editing the search editable field.

**MAS Reference**
https://microsoft.sharepoint.com/teams/msenable/Documents/LCA/MAS%20-%20July%201,%202017%20Release/MAS%202.1.1%20%E2%80%93%20Keyboard.docx?d=w8492c4ebc17940ae8777cd044ed725a2&csf=1

**Suggested fix:**
Applicability: Web  
1- Always ensure each functionality of a page is keyboard accessible. 
E.g. A page with images used as links changes when the user hovers over the image with a mouse. To provide keyboard users with a similar experience, the image is also changed when a user tabs to it 
2-Always try to use standard HTML form controls and link elements to provide keyboard operation and assistive technology interoperability of interactive user interface elements. 
3-Use of both keyboard-specific and mouse-specific events with code that has a scripting function associated with an event. 
Use mousedown with  keydown   
Use mouseup  with keyup   
Use click   with  keypress    
Use mouseover with  focus   
Use mouseout with blur   
e.g. 
<img onclick="nextPage();" onkeypress="nextPage();" tabindex="0" src="arrow.gif"  
alt="Go to next page">  
 
**Key References for Fix:** 
http://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-keyboard-operable.html.

**Refer Attachment**
[MAS 2.1.1_Make code_Advanced control are not accesible.zip](https://github.com/Microsoft/pxt/files/1186644/MAS.2.1.1_Make.code_Advanced.control.are.not.accesible.zip)

[MAS 2.1.1_Make Code_Elements are not accessible.zip](https://github.com/Microsoft/pxt/files/1186640/MAS.2.1.1_Make.Code_Elements.are.not.accessible.zip)

**Comment by veler** [comment]
This is part of Blocky, a third-part component.

**Comment by abchatra** [comment]
We have got exception for Blockly bugs from Julia Liuson (CVP). Closing these as Won't Fix. Following is the reasoning.

----------------------

MakeCode can be used in either Text editor or Block editor.

Blocks editor is inherently visual and is based on google blockly. Usage requires drag and drop functionality and we don’t have a equivalence in MakeCode for this w.r.t accessibility. 
We have evaluated couple of research projects which tries to find a equivalence to block based coding for accessibility needs: 
•	Project Torino (Requires students to buy physical blocks)
•	Accessible blockly (This is text based coding without syntax errors)
Both these are drastic departure from the blockly and provides a complete new experience. These are still in research and we intend not to use them in MakeCode at this point of time.

As workaround, the text editor in MakeCode is the way we see students with the need of accessibility using MakeCode.
